src/math/vector.lisp

```diff
@@ -136,13 +122,13 @@
   4)
 
 (defmacro define-vec-op (name result-type map-op)
-  `(defmethod ,name ((vector1 ,result-type) vector2)
+  `(defmethod ,name ((vector1 ,result-type) (vector2 ,result-type))
      (make-instance ',result-type :in-vec
                     (map 'vector ,map-op
                          (in-vec vector1)
                          (in-vec vector2)))))
 
-(define-vec-op add float2 #'+)
+(define-vec-op add floatn #'+)
 (define-vec-op add float3 #'+)
```

Comment:
Do we still need these methods?

Reply:
Actually yes, because I forgot that we create an instance afterwards, so we need to know the type.